mailpoet/lib/Validator/Validator.php

```diff
@@ -25,11 +25,28 @@ public function __construct(
    * @return mixed
    */
   public function validate(Schema $schema, $value, string $paramName = 'value') {
-    $result = $this->validateAndSanitizeValueFromSchema($value, $schema->toArray(), $paramName);
-    if ($result instanceof WP_Error) {
-      throw ValidationException::createFromWpError($result);
+
+    $schema = $schema->toArray();
+    if (!isset($schema['properties'])) {
+      $result = $this->validateAndSanitizeValueFromSchema($value, $schema, $paramName);
+      if ($result instanceof WP_Error) {
+        throw ValidationException::createFromWpError($result);
+      }
+      return $result;
+    }
+
+    $combinedError = new WP_Error();
+    foreach ($schema['properties'] as $propertyKey => $definition) {
+      $result = $this->validateAndSanitizeValueFromSchema($value, array_merge($schema, ['properties' => [$propertyKey => $definition]]), $paramName);
+      if ($result instanceof WP_Error) {
+        $combinedError->add($propertyKey, $result->get_error_message(), $result->get_error_data());
+      }
+    }
```

Comment:
I think we should also handle `additionalProperties` and `patternProperties`. Or don't they require special treatment? With regards to my comment above (that we could reuse the `mailpoet/lib/Validator/Validator.php` from outside), it may not be needed to solve now.

Maybe we can put this logic back to Validator itself once we're able to make sure all per-object rules work, that we validate all types of properties correctly, etc.? And leave it outside for now?